pkg/machines/libvirt-dbus.js

```diff
@@ -334,7 +334,7 @@ LIBVIRT_DBUS_PROVIDER = {
                     return call(connectionName, storagePoolPath, 'org.freedesktop.DBus.Properties', 'Set', args, TIMEOUT);
                 }, exc => {
                     if (storagePoolPath)
-                        storagePoolUndefine(connectionName, storagePoolPath);
+                        setTimeout(() => storagePoolUndefine(connectionName, storagePoolPath), 1000)
```

Comment:
What I will ask will probably be wrong, because my knowledge of javascript async operations is not good enough, but instead of just waiting 1000ms and hoping that'l enough in all cases, wouldn't it be better to put some lock here? This lock would get locked before Define method is called, and unlocked only after we receive signal VIR_STORAGE_POOL_EVENT_DEFINED, which will result in this function not being called before we receive this signal?
Altought usage of locks in javascript dubious :/ .
WDYT?
If that would not be possible, I think current solution is also good enough.

Reply:
As you say, locks of every kind are not really a thing in javascript. If you have some implementation idea which would solve this nicer I am totally up for it since hardcoded timeouts are really ugly.

Reply:
yeah, hardcoded timeouts are ugly and not really robust. The solution would be to listen for signal (probably `VIR_STORAGE_POOL_EVENT_DEFINED`) and once the signal comes, its callback would do the pool undefining.

Reply:
@marusak Let me explain briefly the VIR_STORAGE_POOL_EVENT_DEFINED and VIR_STORAGE_POOL_EVENT_UNDEFINED because the aforementioned suggestion doesn't really make sense to me.

The VIR_STORAGE_POOL_EVENT_DEFINED is emitted when some pool XML is passed to libvirt (virsh pool-define MYPOOL), so that libvirt is aware of the new pool. The new pool in inactive, so we can't use it yet.

The VIR_STORAGE_POOL_EVENT_UNDEFINED is emmited when we remove from libvirt the knowledge of the pool basically (virsh pool-undefine MYPOOL)

When creating a pool we have the following happening:
* Define the pool the XML we create according to user in-putted values -> DEFINED signal gets emitted, we catch it and in the handler we try to fetch info about the new pool and then add it in the displayed pools in the UI
* Try to activate the pool - this is the step that will fail if the storage configuration has some issue
* If the previous step failed we try to undefine the pool -> UNDEFINE signals gets emitted.

The issue is that sometimes the handler of the UNDEFINE signals gets executed first, removing the pool from the UI, and then the DEFINED signal handler gets executed, adding back the pool in the UI, although it was not supposed to be there.
 
Is that clear? Let me know if I misunderstood your suggestion but calling undefine in the handler of the define signal doesn't really make sense.